SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+		if($ppmEnvironmentalDetails | where { ($_.productType -match "WSA") -and ($_.productAttributes.driftAlarm -match "Red") } ) {
+			$products = $ppmStandardConfigValues | Select-Object -Skip 0 | Where-Object -FilterScript { $_.productType -eq "WSA" }
+			$minCharacterLength = $products.productAttributes.passwdMinimumLength
+			$minLowercaseChar = $products.productAttributes.passwdMinLowercase
+			$minUppercaseChar = $products.productAttributes.passwdMinUppercase
+			$minNumericChar = $products.productAttributes.passwdMinNumeric
+			$minSpecialChar = $products.productAttributes.passwdMinSpecial
+			$passwdHistory = $products.productAttributes.passwdHistoryRestriction
+			$maxConsecutiveIdenticalChar = $products.productAttributes.passwdMaxConsecutiveIdenticalChar
+			$maxPreviousPasswordCharactersReused = $products.productAttributes.passwdMaxPreviousCharactersReused
+			$tempPasswordTtlInHrs = $products.productAttributes.tempPasswdLifetimeInHour
+			$passwordTtlInDays = $products.productAttributes.passwdExpInDays
+			$notificationThresholdInDays = $products.productAttributes.passwdExpReminderInDay
+			$notificationIntervalInDays = $products.productAttributes.passwdExpReminderNotificationFrequencyInDay
+			$numAttempts = $products.productAttributes.passwdMaxFailAttempts
+			$attemptIntervalInMins = $products.productAttributes.passwdAttemptsIntervalInMins
+			$unlockIntervalInMins = $products.productAttributes.passwdUnlockIntervalInMins
+			Write-LogMessage -type INFO -Message "Setting Password Policy on standalone WSA.  Starting..."
+			Start-Sleep -s 10
+			#Write-Host "Requesting WSA Authenication Token" -ForegroundColor Yellow
+			Request-WSAToken -fqdn $ppmVariables.wsaFqdn -user $ppmVariables.wsaAdminUser -pass $ppmVariables.wsaAdminPass | Out-Null
+			#Write-Host "set WSA Password Policy" -ForegroundColor Yellow
+			Set-WSAPasswordPolicy -minLen $minCharacterLength -minLower $minLowercaseChar -minUpper $minUppercaseChar -minDigit $minNumericChar -minSpecial $minSpecialChar -history $passwdHistory -maxConsecutiveIdenticalCharacters $maxConsecutiveIdenticalChar -maxPreviousPasswordCharactersReused $maxPreviousPasswordCharactersReused -tempPasswordTtlInHrs $tempPasswordTtlInHrs -passwordTtlInDays $passwordTtlInDays -notificationThresholdInDays $notificationThresholdInDays -notificationIntervalInDays $notificationIntervalInDays | Out-Null
+			#Write-Host "set WSA Lockout policy" -ForegroundColor Yellow
+			Set-WSAPasswordLockout -numAttempts $numAttempts -attemptInterval $attemptIntervalInMins -unlockInterval $unlockIntervalInMins | Out-Null
+			Write-LogMessage -type INFO -Message "Setting Password Policy on standalone WSA.  Complete..."
+		}
+	} else {
+		Write-LogMessage -Type INFO -Message "Workspace One Access Skipped.  Login details are not provided."  
+	}
+}
+
+# MAIN FUNCTION
+try {
+	if ((($publishHTML -or $publishJSON) -and $configurePasswordPolicy) -or ($publishHTML -and $publishJSON)) {
```

Comment:
You can handle this without code by using  ParameterSetName = 'value' against the input parameter, this way if you select one then anything with not the same ParameterSetName  is ignored on the command line

Reply:
Agreed.

Reply:
I am not sure how to do this.  but will look into it along with other improvement mentioned above.

Reply:
per conversation with Ryan, push to future enhancement